dao/mongo.go

```diff
@@ -273,3 +274,37 @@ func (m *MongoService) GetPaymentsWithRefundStatus() ([]models.PaymentResourceDB
 
 	return payments, nil
 }
+
+// GetPaymentsWithPaidStatus retrieves a list of all payments in the DB with a status of PAID
+func (m *MongoService) GetPaymentsWithPaidStatus() ([]models.PaymentResourceDB, error) {
+	var payments []models.PaymentResourceDB
+
+	collection := m.db.Collection(m.CollectionName)
+	dataStatusFilter := bson.M{paymentStatus: "paid"}
+	refundStatusFilter := bson.M{refundStatus: bson.M{"$eq": "pending"}}
+	statusFilter := bson.M{"$and": bson.A{dataStatusFilter, refundStatusFilter}}
+
+	paymentDBResources, err := collection.Find(context.Background(), statusFilter)
+	if err != nil {
+		return nil, err
+	}
+
+	for paymentDBResources.Next(context.Background()) {
```

Comment:
Would this work instead of a loop?
```suggestion
	err = paymentDBResources.All(context.Background(), &payments)
	if err != nil {
		return nil, err
	}
```

Reply:
I used loop because it is more reliable than 'All' while dealing with large volume of data. However, i see your point and for the sake of uniformity will change to 'All', if need be could be amended later.Thanks